pulsar-client-cpp/docker-tests.sh

```diff
@@ -58,4 +60,26 @@ done
 
 # Start 2 Pulsar standalone instances (one with TLS and one without)
 # and execute the tests
-$DOCKER_CMD bash -c "cd /pulsar/pulsar-client-cpp && ./run-unit-tests.sh ${tests}"
+set +e
+DISABLE_COLOR_OUTPUT=""
+if [ "$GTEST_COLOR" = "no" ]; then
+  DISABLE_COLOR_OUTPUT="| cat"
+fi
+$DOCKER_CMD bash -c "cd /pulsar/pulsar-client-cpp && ./run-unit-tests.sh ${tests} $DISABLE_COLOR_OUTPUT"
```

Comment:
The real cause of broken CI is here. `DISABLE_COLOR_OUTPUT` is `| cat` by default, then the command becomes

```bash
$DOCKER_CMD bash -c "cd /pulsar/pulsar-client-cpp && ./run-unit-tests.sh ${tests} | cat"
```

The exit code is what `cat` exits, so it's always 0. However, if you changed it to

```bash
$DOCKER_CMD bash -c "cd /pulsar/pulsar-client-cpp && ./run-unit-tests.sh ${tests}"
```

The exit code would become 1, which is what `./run-unit-tests.sh` exits.

Reply:
@BewareMyPower yes that's true that `| cat` causes the issue. it would require `set -o pipefail;` for the exit code to propagate when there's `| cat` .

Reply:
fixed by PR #11575